models/fivetran_platform__connector_status.sql

```diff
@@ -1,28 +1,38 @@
 with transformation_removal as (
 
-    select *
+    select 
+        *,
+        case when event_subtype in ('status', 'sync_end')
+            then message_data
+            else null end as filtered_message_data
     from {{ ref('stg_fivetran_platform__log') }}
     where transformation_id is null
+),
 
+parse_json as (
+    select
+        *,
+        {{ fivetran_log.fivetran_log_json_parse(string="filtered_message_data", string_path=["status"]) }} as log_status,
+        {{ fivetran_log.fivetran_log_json_parse(string="filtered_message_data", string_path=["reason"]) }} as log_reason
+    from transformation_removal
```

Comment:
Added this cte to handling all parsing needs in one place. Intention to make this easier to maintain and read.